Minor refinements in syntax

hideya opened the pull request:
Thanks for creating and sharing the great example!
Reading the code, I found places where I think we can simplify a bit.  Although I touched many lines, all the changes are local and simple (usage of trailing closures, simplifications of variable usages, etc).  Hope those modifications make sense...

By the way, there is a different approach when to manipulate `Widget` model in `LBRepositoryController`'s `createModel()`, `updateModel()` and `deleteModel()`.  I should say there is too little information about LoopBack iOS SDK usage in Swift...  I have a simple example code and am thinking to upload it to github for your reference (I need some cleanup work before that).  I'll give you a ping when I'm done!

hideya committed:
Minor refinements in syntax

hideya commented:
@kgoedecke I've uploaded a simple swift example here: https://github.com/hideya/loopback-swift-simple-example

As you see in the code, if we create a new model like `let newWidget = repository.modelWithDictionary(nil) as! Widget`, we can call `newWidget.saveWithSuccess(...)` directly. I thought you might want to check out this approach.  I guess I better submit an Issue about it for a discussion.  Hope it helps.

kgoedecke commented:
@hideya Thanks I looked into it. I think the major difference between our implementations is that you call AppDelegate.coffeeShopRepo.allWithSuccess every time you refresh the table. I was trying to avoid this as it's not really efficient especially if you have images in your tables at one point. But my approach leads to the problem that I mentioned here: https://github.com/kgoedecke/loopback-swift-crud-example/issues/2, maybe you have any ideas on the ._id thing.

hideya commented:
@kgoedecke thanks for checking out the code.  I see.  I kept the code in that way for the sake of simplicity (I basically followed [this tutorial](https://strongloop.com/strongblog/how-to-crud-ios-nodejs-loopback-api-server-part-1/) when to create the app).  I'll look into the `._id` issue you pointed out in https://github.com/kgoedecke/loopback-swift-crud-example/issues/2

hideya commented:
@kgoedecke I've updated the simple example not to invoke `AppDelegate.coffeeShopRepo.allWithSuccess` when to refresh the table.  Now it updates the table contents locally just like you do in your app.  Since I didn't encounter the `._id` issue on my side, I added Swipe-to-delete feature to try out model deletion.  It seems working fine.  I'll look into your app when I get time next.

kgoedecke commented:
@hideya I'm starting to think your implementation is the proper way to go. I might adjust my repository later today! Thanks for all the input, I really appreciate it and hope more people will start looking into LoopBack Swift development.
